htdocs/api/v0.0.3-dev/candidates/visits/dicoms/Dicom.php

```diff
@@ -0,0 +1,147 @@
+ */
+class Dicom extends \Loris\API\Candidates\Candidate\Visit
+{
+    /**
+     * Construct a Dicom class object to output the candidate visits dicom tar file
+     *
+     * @param string $method     The method of the HTTP request
+     * @param string $CandID     The CandID to be serialized
+     * @param string $VisitLabel The visit label to be serialized
+     * @param string $Tarname    The Tar name to be retrieved
+     */
+    public function __construct($method, $CandID, $VisitLabel, $Tarname)
+    {
+        $requestDelegationCascade = $this->AutoHandleRequestDelegation;
+
+        $this->AutoHandleRequestDelegation = false;
+
+        if (empty($this->AllowedMethods)) {
+            $this->AllowedMethods = ['GET'];
+        }
+        $this->CandID     = $CandID;
+        $this->VisitLabel = $VisitLabel;
+        $this->Tarname    = $Tarname;
+
+        parent::__construct($method, $CandID, $VisitLabel);
+
+        $results =  $this->getFullPath();
+        if (empty($results)) {
+            $this->header("HTTP/1.1 404 Not Found");
+            $this->error("File not found");
+            $this->safeExit(0);
+        }
+
+        if ($requestDelegationCascade) {
+            $this->handleRequest();
+        }
+
+    }
+
+    /**
+     * Handles a GET request
+     *
+     * @return none, but send the file content to stdout
+     */
+    public function handleGET()
+    {
+        $fullDir = $this->getFullPath();
+        $fp      = fopen($fullDir, "r");
+        if ($fp !== false) {
+            $this->Header("Content-Type: application/x-tar");
+            $this->Header('Content-Length: '.filesize($fullDir));
+            $this->Header('Content-Disposition: filename='.$this->Tarname);
+            fpassthru($fp);
+            fclose($fp);
+            $this->safeExit(0);
+        } else {
+            $this->header("HTTP/1.1 500 Internal Server Error", true, 500);
+            $this->error("Could not load Tarfile");
+            $this->safeExit(1);
+        }
+    }
+
+    /**
+     * Gets the full path of this image on the filesystem, in order
+     * to be able to pass it to an fopen command (or similar)
+     *
+     * @return string
+     */
+    protected function getFullPath()
+    {
+        return $this->getTarchivePath() . "/" . $this->getTarFile();
+    }
+
+    /**
+     * Gets the root of the tarchive directory, so that we know where
+     * to retrieve images relative to.
+     *
+     * @return string
+     */
+    protected function getTarchivePath()
+    {
+        $factory = \NDB_Factory::singleton();
+        $config  = $factory->Config();
+        return $config->getSetting("tarchiveLibraryDir");
+    }
+
+    /**
+     * Gets the Tarname saved in the database for this file
+     *
+     * @return string
+     */
+    protected function getTarFile()
+    {
+        $factory      = \NDB_Factory::singleton();
+        $db           = $factory->Database();
+        $partial_path = $db->pselectRow(
```

Comment:
could be slightly simplified with `pselectOne`

Reply:
Ok!